src/jdk.incubator.jextract/share/classes/jdk/internal/jextract/impl/MacroParserImpl.java

```diff
@@ -81,16 +81,24 @@ public MacroParserImpl(TranslationUnit tu, Collection<String> args) {
                     Optional.of((Macro)result) :
                     Optional.empty();
         } catch (Throwable ex) {
+            // This ate the NPE and cause skip of macros
+            // Why are we expecting exception here? Simply be defensive?
+            if (JextractTaskImpl.VERBOSE) {
+                System.err.println("Failed to handle macro " + macroName);
+                ex.printStackTrace(System.err);
+            }
             return Optional.empty();
         }
     }
 
     MacroResult reparse(String snippet) {
-        return reparser.reparse(snippet)
+        MacroResult rv = reparser.reparse(snippet)
                 .filter(c -> c.kind() == CursorKind.VarDecl &&
                         c.spelling().contains("jextract$"))
                 .map(c -> compute(c))
                 .findAny().get();
+        typeMaker.resolveTypeReferences();
```

Comment:
Should this call go here? Probably - if we drop the toplevel unit after subsequent reparse. But, I think we should also put a call on this at the end of parsing? Let's say I have  something like:

```
struct A {
   struct B *b;
}

struct B {
  struct A *a;
}
```

No macros here - how does this work? I believe you will add pointers in the todo list (e.g. when we parse `struct A` we will add `B*`). Who resolves that? And, if we did a resolve, wouldn't that end up adding another pointer (`A*`) to the list? Who would resolve that? And if we re-run resolve on the new pointer, how do we prevent the logic from re-adding the already resolved pointer type `B*` ?

Reply:
resolveTypeReferences suppose to be called after parse or reparse, so we no longer depends on clang type.

Theoretically, we also should call resolveTypeReference after parse for non-macro case, but as I noted in the code, we haven't completely get rid of dependency on cursor for declaration API, so we now still have to operate under an valid clang TU for non-macro case, thus there is no need to call that just yet. Because when type() is called on the pointer, the resolve happens automatically.